File: spark/src/main/scala/geotrellis/spark/io/file/FileLayerHeader.scala
package geotrellis.spark.io.file

import geotrellis.spark.io.LayerHeader
import spray.json._

case class FileLayerHeader(
  keyClass: String,
  valueClass: String,
  path: String
) extends LayerHeader {
  def format = "file"
}

object FileLayerHeader {
  implicit object FileLayerHeaderFormat extends RootJsonFormat[FileLayerHeader] {
    def write(md: FileLayerHeader) =
      JsObject(
        "format" -> JsString(md.format),
        "keyClass" -> JsString(md.keyClass),
        "valueClass" -> JsString(md.valueClass),
        "path" -> JsString(md.path)
      )

    def read(value: JsValue): FileLayerHeader =
      value.asJsObject.getFields("keyClass", "valueClass", "path") match {
        case Seq(JsString(keyClass), JsString(valueClass), JsString(path)) =>
          FileLayerHeader(
            keyClass,
            valueClass,
            path
          )

        case _ =>
          throw new DeserializationException(s"FileLayerHeader expected, got: $value")
      }
  }
}

File: geotrellis-python/geotrellis/spark/io/file/FileLayerHeader.py
from __future__ import absolute_import
from geotrellis.spark.io.LayerHeader import LayerHeader
from geotrellis.python.util.utils import JSONFormat

class FileLayerHeader(LayerHeader):
    implicits = {'format': lambda: FileLayerHeaderFormat()}
    def __init__(self, key_class, value_class, path):
        self._key_class = key_class
        self._value_class = value_class
        self.path = path
    @property
    def format(self):
        return 'file'
    @property
    def keyClass(self):
        return self._key_class
    @property
    def valueClass(self):
        return self._value_class

class FileLayerHeaderFormat(JSONFormat):
    def to_dict(self, obj):
        return {"format": obj.format,
                "keyClass": obj.keyClass,
                "valueClass": obj.valueClass,
                "path": obj.path}
    def from_dict(self, dct):
        fields = self.get_fields(dct, 'keyClass', 'valueClass', 'path')
        if not fields:
            raise DeserializationException("FileLayerHeader expected")
        key_class, value_class, path = fields
        return FileLayerHeader(key_class, value_class, path)

